src/DGtal/geometry/curves/AlphaThickSegmentComputer.ih

```diff
@@ -0,0 +1,965 @@
+    return;   /* skip an interior vertex */
+  else //if( IsLeft( D[0], D[1], P ) < 0 || IsLeft( D[n-2], D[n-1], P ) < 0 )
+    {
+      while( melkmanIsLeft( myState.melkmanQueue[0], myState.melkmanQueue[1], aPoint ) <= 0 )
+        myState.melkmanQueue.pop_front();    /* remove bottom of deque */
+
+      myState.melkmanQueue.push_front( aPoint );  /* insert id at bottom of deque */
+
+      while( melkmanIsLeft( myState.melkmanQueue[int(myState.melkmanQueue.size())-2], 
+                            myState.melkmanQueue[int(myState.melkmanQueue.size())-1], aPoint ) <= 0 )
+        myState.melkmanQueue.pop_back();     /* pop top of deque */
+
+      myState.melkmanQueue.push_back( aPoint );   /* push id onto top of deque */
+    }
+}
+
+
+
+template < typename TInputPoint, typename TConstIterator>
+inline
+void
+DGtal::AlphaThickSegmentComputer< TInputPoint, TConstIterator>::
+melkmanInit() 
+{
+  InputPoint aPoint1 = myState.melkmanQueue[0];
+  InputPoint aPoint2 = myState.melkmanQueue[1];
+  InputPoint aPoint3 = myState.melkmanQueue[2];
+  myState.melkmanQueue.clear();
+
+  if( melkmanIsLeft( aPoint1, aPoint2, aPoint3 ) == 0.0 )
+    {
+      myState.melkmanQueue.push_back(aPoint3);
+      myState.melkmanQueue.push_back(aPoint1);
+      myState.melkmanQueue.push_back(aPoint3);
+    }
+  else
+    {
+      myState.melkmanQueue.push_front(aPoint3);
+      /* ccw vertices are: 2,0,1,2 */
+      if( melkmanIsLeft( aPoint1, aPoint2, aPoint3 ) > 0 )    
+        {
+          myState.melkmanQueue.push_back(aPoint1);
+          myState.melkmanQueue.push_back(aPoint2); 
+        }
+      else /* ccw vertices are: 2,1,0,2 */
+        {
+          myState.melkmanQueue.push_back(aPoint2);
+          myState.melkmanQueue.push_back(aPoint1); 
+        }  
+      myState.melkmanQueue.push_back(aPoint3);
+    }   
+}
+
+
+
+
+
+//-----------------------------------------------------------------------------
+template < typename TInputPoint, typename TConstIterator>
+inline
+bool
+DGtal::AlphaThickSegmentComputer< TInputPoint, TConstIterator>::melkmanIsConvexValid() 
+{
+  if (!myState.isMelkmanInitialized)
+    {
+      return true;
+    } 
+  myState.actualThickness = melkmanUpdateMainHeight(); 
+  return myState.actualThickness <= myMaximalThickness;
+}
+
+
+
+//-----------------------------------------------------------------------------
+template < typename TInputPoint, typename TConstIterator>
+inline
+typename TInputPoint::Component
+DGtal::AlphaThickSegmentComputer< TInputPoint,
+                                  TConstIterator>::melkmanIsLeft(const InputPoint &aPointIndex0,
+                                                                 const InputPoint &aPointIndex1,
+                                                                 const InputPoint &aPointIndex2) const
+{
+  return ((aPointIndex1 - aPointIndex0)[0] *  (aPointIndex2 - aPointIndex0)[1])   -
```

Comment:
This is exactly the method "Orientation" I was talking about above. Something that say whether for 3 points P,Q,R, R is to the left of vector PQ (determinant is positive), on or to the right.